docs/tests.md

```diff
@@ -37,6 +38,78 @@ You can run a subset of tests which match a `description` or `it` with:
+another tab/window. This is the most common cause of race conditions that I find in tests. webdriver automatically
+switches contexts to any new "window" that is created. chromedriver can’t differentiate between tabs and windows in
+brave which is why we have special helpers. If there is more than one tab with the same url, you can’t use waitForUrl:
+it will return on the tab/window that already exists and not wait for the new one. `getTabCount` is a good alternative
+because it doesn’t require a particular window context to run
+
+- Don’t try to interact with any dom element (click, moveTo, etc…) until you have verified that it exists.
+Verifying the page or some parent element is not sufficient
+
+- Avoid using external sites unless it’s **absolutely necessary**. For example where we need to do this: the httpse and
+other ssl related tests make external calls because we don’t current have ssl capabilities on the test server.
+
+  Adding an external dependency causes these issues:
+    - tests will run slower since external requests need to be made
+    - tests are not runnable if network is down
+    - if the site's content changes, the test will break
+
+- Never assume your tests will run in a particular order. Before each test is run, you should do the setup or cleanup
+required in order for the test to run properly.
+
+- Where it makes sense, prefer to use a before() instead of beforeEach(). Because the webdriver initialization takes
+a while, having it set up once and being able to run multiple tests is going to help speed things up. Please note that
+this isn't always an option though, because the test might alter the environment in a way that you need to redo the setup
+(in which case a beforeEach() OR moving that test into it's own describe()/before() is a better choice).
+
+## Useful helper methods
+
+These are methods we own which can be used along side the [existing web driver methods](http://webdriver.io/api.html).
+Please note that it's fairly easy to add new helper methods. For more info, please check out the addCommands()
+method in at [brave.js](https://github.com/brave/browser-laptop/blob/master/test/lib/brave.js).
+
+### IPC
+
+- ipcSend: send an IPC message, like SHORTCUT_NEW_FRAME, etc. To see more message types, check out componentDidMount in [main.js](https://github.com/brave/browser-laptop/blob/master/js/components/main.js)
+- ipcSendRenderer
+
+### Tab management
+
+- tabHandles
+- tabByIndex
+- getTabCount
+- tabByUrl
+
+### Window management
+- waitForBrowserWindow
+- setContextMenuDetail: hides any context menus which are actively showing
+- showFindbar
+- setPinned
+- getDefaultWindowHeight
+- getDefaultWindowWidth
+- resizeWindow
+- windowParentByUrl
+- windowByUrl
+
+### State management
+- getAppState
+- getWindowState
+
+### Site management
+- addSite: add a bookmark/folder or history entry
+- addSiteList
+- removeSite
+
+### Preferences
+- changeSetting
+- changeSiteSetting
```

Comment:
agreed, we should wait for the change on these so they don't need to be in every test.